File: python/Code/Instructor.py
from User import User
from abc import ABC, abstractmethod
class Instructor(User):
    # Konstruktor untuk membuat objek Instructor baru
    def __init__(self, userId, name, expertise):
        # Memanggil konstruktor dari kelas induk (User)
        super().__init__(userId, name)
        self.__expertise = expertise

    # Mengambil keahlian instruktur
    def getExpertise(self):
        return self.__expertise

    # Mengubah keahlian instruktur
    def setExpertise(self, expertise):
        self.__expertise = expertise

    # Menampilkan informasi detail instruktur (override dari kelas User)
    def displayInfo(self):
        print(f"Instructor: {self._name} ({self._id}) | Expertise: {self.__expertise}")

File: java/Code/Instructor.java
// Kelas turunan dari User untuk merepresentasikan data seorang instruktur (Instructor)
public class Instructor extends User {
    // Atribut
    private String expertise; // Keahlian khusus instruktur

    // Konstruktor untuk membuat objek Instructor baru
    public Instructor(String userId, String name, String expertise) {
        // Memanggil konstruktor dari kelas induk (User)
        super(userId, name);
        this.expertise = expertise;
    }

    // Mengambil keahlian instruktur
    public String getExpertise() {
        return expertise;
    }

    // Mengubah keahlian instruktur
    public void setExpertise(String expertise) {
        this.expertise = expertise;
    }

    // Menampilkan informasi detail instruktur (override dari kelas User)
    @Override
    public void displayInfo() {
        System.out.println("Instructor ID: " + this.userId + ", Name: " + this.name + ", Expertise: " + this.expertise);
    }
}
